### DISCUSSION TITLE
Q: Web access possible without Application Service and subdomains?

### DISCUSSION THREAD

**Discussion by furrnace** [gravitational/teleport#25400] [Q&A]
From the documentation it seems like I need to install teleport as application service on the host where the application is provided, or on a closeby host. 

Many of the web applications I want to access are appliances where I either cannot install anything (e.g. APC UPS NMC), or do not want to install anything (e.g. TrueNAS Core, pfSense plus). As those applications reside in the management VLAN, the teleport proxy host is a closeby as any other host. So it seems I would start the teleport application service also on the proxy host. However, this seems rather useless, as it would just create communication overhead, and would not benefit security at all.

So is there a way to let the teleport proxy make Web applications accessible without connecting to a Teleport Application Service? Would that be possible without providing subdomains for each application service?

**Comment by webvictim** [comment]
If you want to make applications available through Teleport, it's mandatory to run the `app_service` _somewhere_. That can be on a separate close-by host, or you can run `auth_service`, `proxy_service` and `app_service` all on the same host in one Teleport process if you like by enabling them in `/etc/teleport.yaml`.

You can also set any domain or subdomain that you like as the public address (`public_addr`) for an application behind Teleport - it doesn't necessarily have to be a subdomain of the Teleport proxy's address.

The only requirements for this to work are that you point the DNS record for any domain you use to the Teleport proxy's IP, and that Teleport either has ACME support enabled, or has a valid TLS certificate loaded for each domain via the `https_keypairs` list in the config file.

**Reply by furrnace** [reply]
would it still work if I just were using a valid certificate for the teleport proxy host and run auth, proxy and app service on the same machine? That would mean that all services are represented through the same dns name and certificate?

(I know this doesn't make sense in enterprise production, but I am just having a home lab with a dozen services in the management VLAN)
